### PR TITLE
feat: Create code of conduct rest api

### PR DESCRIPTION
**Pull request by nverbic**
Description:

Create backend rest api for Code of conduct page.
Closes https://github.com/Women-Coding-Community/wcc-backend/issues/12

Add CodeOfConductPage.java
  - public record CodeOfConductPage(Page page, List<Section> items)
Add Section.java
  - public record Section(String title, String description, List<String> items)
Add codeOfConductPage.json
  - Json file with data to be sent in the response to the frontend
Modified AboutController.java
  - public CodeOfConductPage getCodeOfConductPage() with GET mapping
Modified CmsService.java
  - public CodeOfConductPage getCodeOfConduct() to retrieve Code of conduct
 
Add unit tests and integration tests for CodeOfConduct API implementation.

Screenshot (82)

Backend json response:

Screenshot (85)

Add JavaDocs for service and controller functions

Screenshot (88)

Screenshot (87)

### CODE REVIEW

File: src/test/resources/example.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Line 1
```
Comment (jpmaHub) [inline src/test/resources/example.txt:1]: do we need to commit this example file text to github repo?

File: src/main/resources/codeOfConductPage.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (dricazenck) [inline src/main/resources/codeOfConductPage.json:1]: Same as previous PR, you should create the current code of conduct json from our website. This file, for example you can copy for test.

File: src/main/java/com/wcc/platform/controller/AboutController.java [hunk 3]
```diff
@@ -27,4 +28,10 @@ public AboutController(CmsService service) {
     public ResponseEntity<TeamPage> getTeamPage() {
         return ResponseEntity.ok(cmsService.getTeam());
     }
+
+    @GetMapping("/code-of-conduct")
+    @ApiOperation(value = "API to retrieve Code of Ccnduct")
+    public ResponseEntity<CodeOfConductPage> getCodeOfConductPage() {
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/controller/AboutController.java:34]: Java doc here, and test please.

File: src/main/java/com/wcc/platform/service/CmsService.java [hunk 4]
```diff
@@ -35,4 +37,18 @@ public TeamPage getTeam() {
             throw new PlatformInternalException(e.getMessage(), e);
         }
     }
+
+    /**
+     * API to retrieve Code of conduct.
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/service/CmsService.java:42]: Sorry, my bad, this javadoc about API should be in the controller. 

For the service, we should talk what is doing here, for example: 

```suggestion
     * Read JSON and convert to Pojo CodeOfConductPage 
```

File: src/main/java/com/wcc/platform/service/CmsService.java [hunk 5]
```diff
@@ -35,4 +37,18 @@ public TeamPage getTeam() {
...
+     *
+     * @return Code of conduct page content.
+     */
+    public CodeOfConductPage getCodeOfConduct() {
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/service/CmsService.java:46]: Also please create unit test for that logic.

File: src/main/java/com/wcc/platform/service/CmsService.java (hunk 4 above)
Comment (nverbic) [reply src/main/java/com/wcc/platform/service/CmsService.java]: @goelsonali @dricazenck I updated also for footer controller and service functions.

File: src/main/java/com/wcc/platform/controller/FooterController.java [hunk 6]
```diff
@@ -24,6 +24,11 @@ public FooterController(CmsService cmsService) {
         this.cmsService = cmsService;
     }
 
+    /**
+     * API to retrieve footer section details.
+     *
+     * @return Footer content.
+     */
     @GetMapping("/footer")
```
Comment (dricazenck) [inline src/main/java/com/wcc/platform/controller/FooterController.java:32]: ❤️

File: src/test/java/com/wcc/platform/service/CmsServiceIntegrationTest.java [hunk 7]
```diff
@@ -65,4 +67,19 @@ void getCollaboratorPageTest() {
...
+
+        var expectedCodeOfConductPage = createCodeOfConductPageTest(CODE_OF_CONDUCT.getFileName());
+
+        assertEquals(expectedCodeOfConductPage.page(), result.page());
+        
+        assertEquals(2, result.items().size());
+        assertNotNull(result.items().get(0).title());
+        assertNotNull(result.items().get(0).description());
+        assertEquals(5, result.items().get(0).items().size());
+        assertNotNull(result.items().get(0).items().get(0));
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/service/CmsServiceIntegrationTest.java:83]:
```suggestion


        assertEquals(expectedCodeOfConductPage, result);
```

File: src/test/java/com/wcc/platform/controller/AboutControllerTest.java [hunk 8]
```diff
@@ -121,4 +123,50 @@ void testCollaboratorOKResponse() throws Exception {
                         .andExpect(jsonPath("$.collaborators[0].network[0].type", is("LINKEDIN")))
                         .andExpect(jsonPath("$.collaborators[0].network[0].link", is("collaborator_link")));
     }
+
+    @Test
+    void testCodeOfConductNotFound() throws Exception {
+        when(service.getCodeOfConduct()).thenThrow(new ContentNotFoundException("Not Found Exception"));
+
+        mockMvc.perform(get("/api/cms/v1/code-of-conduct")
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/controller/AboutControllerTest.java:131]: you could extract to a constant this string and reuse in the test.

```suggestion
      private static final String API = "/api/cms/v1/code-of-conduct";
      
      .... 
      
      
      mockMvc.perform(get(API)
```

File: src/test/java/com/wcc/platform/controller/AboutControllerTest.java [hunk 9]
```diff
@@ -121,4 +123,50 @@ void testCollaboratorOKResponse() throws Exception {
...
+                        .contentType(APPLICATION_JSON))
+                .andExpect(status().isInternalServerError())
+                .andExpect(jsonPath("$.status", is(500)))
+                .andExpect(jsonPath("$.message", is("internal Json")))
+                .andExpect(jsonPath("$.details", is("uri=/api/cms/v1/code-of-conduct")));
+    }
+
+    @Test
+    void testCodeOfConductOKResponse() throws Exception {
+        var codeOfConductPage = new CodeOfConductPage(
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/controller/AboutControllerTest.java:154]: I did like this for example in my PR: 

```
    @Test
    void testOKResponse() throws Exception {
        var fileName = MENTORSHIP.getFileName();
        var expectedJson = FileUtil.readFileAsString(fileName);

        when(service.getOverview()).thenReturn(createMentorshipPageTest(fileName));

        mockMvc.perform(get(API_MENTORSHIP_OVERVIEW).contentType(APPLICATION_JSON))
                .andExpect(status().isOk())
                .andExpect(content().json(expectedJson));
    }

```

File: src/test/java/com/wcc/platform/controller/AboutControllerTest.java [hunk 10]
```diff
@@ -121,4 +123,50 @@ void testCollaboratorOKResponse() throws Exception {
...
+            new Page("code_of_conduct_title", "code_of_conduct_subtitle", "code_of_conduct_desc"),
+            List.of(new Section("section_title", "section_description", List.of("item_1", "item_2", "item_3"))));
+
+        when(service.getCodeOfConduct()).thenReturn(codeOfConductPage);
+
+        mockMvc.perform(get("/api/cms/v1/code-of-conduct")
+                        .contentType(APPLICATION_JSON))
+                        .andExpect(status().isOk())
+                        .andExpect(jsonPath("$.page.title", is("code_of_conduct_title")))
```
Comment (dricazenck) [inline src/test/java/com/wcc/platform/controller/AboutControllerTest.java:163]:
```suggestion
      .andExpect(content().json(expectedJson));
```

**Review by dricazenck — APPROVED** [review]
🚀  Very nice!!

**Pull request merged**